src/test/java/redis/clients/jedis/tests/JedisPoolTest.java

```diff
@@ -381,4 +382,41 @@ private int getClientCount(final String clientList) {
     return clientList.split("\n").length;
   }
 
+  @Test
+  public void testResetInvalidPassword() {
+    JedisFactory factory = new JedisFactory(hnp.getHost(), hnp.getPort(), 2000, 2000, "foobared", 0, "my_shiny_client_name", false, null, null, null);
+    JedisPool pool = new JedisPool(new JedisPoolConfig(), factory);
+    Jedis obj1 = pool.getResource();
+    obj1.set("foo", "bar");
+    assertEquals("bar", obj1.get("foo"));
+    assertEquals(1, pool.getNumActive());
+    obj1.quit();
```

Comment:
Use `close()`

Reply:
When using close(),  the factory.setPassword() doesn't take any effect in this case since  makeObject() of JedisFactory.java doesn't get invoked so the same Jedis object returned with the old password.

Reply:
Then remove `quit()`